mantle/platform/api/azure/storage.go

```diff
@@ -15,29 +16,176 @@
 package azure
 
 import (
+	"context"
 	"fmt"
+	"io"
+	"os"
 	"strings"
 
-	"github.com/Azure/azure-sdk-for-go/arm/storage"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/streaming"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/to"
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/storage/armstorage"
+	"github.com/Azure/azure-sdk-for-go/sdk/storage/azblob"
+	"github.com/Azure/azure-sdk-for-go/sdk/storage/azblob/blob"
+	"github.com/Azure/azure-sdk-for-go/sdk/storage/azblob/bloberror"
+	"github.com/Azure/azure-sdk-for-go/sdk/storage/azblob/pageblob"
+
+	"github.com/frostschutz/go-fibmap"
 )
 
-func (a *API) GetStorageServiceKeys(account, resourceGroup string) (storage.AccountListKeysResult, error) {
-	return a.accClient.ListKeys(resourceGroup, account)
+func (a *API) GetStorageServiceKeys(account, resourceGroup string) (armstorage.AccountListKeysResult, error) {
+	resp, err := a.accClient.ListKeys(context.Background(), resourceGroup, account, &armstorage.AccountsClientListKeysOptions{Expand: nil})
+	if err != nil {
+		return armstorage.AccountListKeysResult{}, err
+	}
+	return resp.AccountListKeysResult, nil
 }
 
 func (a *API) CreateStorageAccount(resourceGroup string) (string, error) {
 	// Only lower-case letters & numbers allowed in storage account names
 	name := strings.Replace(randomName("kolasa"), "-", "", -1)
-	parameters := storage.AccountCreateParameters{
-		Sku: &storage.Sku{
-			Name: "Standard_LRS",
+	parameters := armstorage.AccountCreateParameters{
+		SKU: &armstorage.SKU{
+			Name: to.Ptr(armstorage.SKUNameStandardLRS),
 		},
-		Kind:     "Storage",
-		Location: &a.opts.Location,
+		Kind:     to.Ptr(armstorage.KindStorage),
+		Location: to.Ptr(a.opts.Location),
 	}
-	_, err := a.accClient.Create(resourceGroup, name, parameters, nil)
+	_, err := a.accClient.BeginCreate(context.Background(), resourceGroup, name, parameters, nil)
```

Comment:
Presumably the caller expects the storage account to be available immediately upon return, so we should block until the operation is complete.

Reply:
indeed. I'll push up another revision once you answer https://github.com/coreos/coreos-assembler/pull/3349/files#r1109339408